core/src/test/java/de/danoeh/antennapod/core/storage/DbReaderTest.java

```diff
@@ -290,32 +291,68 @@ public void testGetNewItemIds() {
 
     @Test
     public void testGetPlaybackHistory() {
-        final int numItems = (DBReader.PLAYBACK_HISTORY_SIZE + 1) * 2;
-        final int playedItems = DBReader.PLAYBACK_HISTORY_SIZE + 1;
-        final int numReturnedItems = Math.min(playedItems, DBReader.PLAYBACK_HISTORY_SIZE);
-        final int numFeeds = 1;
+        // This for loop that manually calls setUp() and tearDown() is a hack
+        // because it seems that supporting a single parameterized test in a
+        // Robolectric test suite like this is difficult or impossible.
```

Comment:
If that makes things easier, feel free to create a new parameterized test class for this. Manually calling `setUp` is indeed a bit hacky.

Reply:
OK, this was actually easier than I thought, I refactored this test class into two enclosing inner classes, one parameterized and one not parameterized, with shared `setUp` and `tearDown` methods.